enhancements/dns/externaldns-operator.md

```diff
@@ -0,0 +1,467 @@
+DNS provider). The Ingress Controller API could be expanded to include the ability for a cluster administrator to specify whether or not a Wildcard DNSRecord
+is desired.
+
+
+Here is an example of what an ExternalDNS deployment (created by the operator) would look like:
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: external-dns
+spec:
+  strategy:
+    type: Recreate
+  selector:
+    matchLabels:
+      app: external-dns
+  template:
+    metadata:
+      labels:
+        app: external-dns
+    spec:
+      serviceAccountName: external-dns
+      containers:
+      - name: external-dns-private
+        image: k8s.gcr.io/external-dns/external-dns:v0.8.0
+        args:
+        - --source=openshift-route
+        - --provider=aws
+        - --policy=upsert-only # would prevent ExternalDNS from deleting any records, omit to enable full synchronization
+        # can only have one zone at a time if domains are overlapping over zones
+        - --aws-zone-type=private
+        - --registry=txt
+        - --txt-owner-id=externalDNS-operator
+        - --log-level=debug
+        # Needs to be unique per container
+        - --metrics-address=:7979
+        - --zone-id-filter=/hostedzone/myzone
+        env:
+          # Base64 decode from cloud-credentials secret generated by cloud credentials request
+          - name: AWS_ACCESS_KEY_ID
+            value: abc
+          - name: AWS_SECRET_ACCESS_KEY
+            value: xyz
+      - name: external-dns-public
+        image: k8s.gcr.io/external-dns/external-dns:v0.8.0
+        args:
+        - --source=openshift-route
+        - --provider=aws
+        - --policy=upsert-only # would prevent ExternalDNS from deleting any records, omit to enable full synchronization
+        # can only have one zone at a time if domains are overlapping over zones
+        - --aws-zone-type=public
+        - --registry=txt
+        - --txt-owner-id=my-identifier
+        - --log-level=debug
+        # Needs to be unique per container
+        - --metrics-address=:7980
+        # Filter from aws dev account
+        - --zone-id-filter=/hostedzone/myzone
+        env:
+          # Base64 decode from cloud-credentials secret generated by cloud credentials request
+          - name: AWS_ACCESS_KEY_ID
+            value: abc
+          - name: AWS_SECRET_ACCESS_KEY
+            value: xyz
+```
+
+Note the use of multiple ExternalDNS containers within the same pod int he above deployment. Metrics listening ports would have
+to be configured on a per-container basis to avoid collisions.
+
+Metrics will be exposed via an insecure channel by default.
+On OpenShift, externalDNS instances can be deployed with a reverse proxy sidecar, such as kube-rbac-proxy, to provide a secure means of metrics scraping.
+
+### Risks and Mitigations
+
+Cluster administrators may not want to give users the ability to create arbitrary DNS records in an external provider when they grant users permission to create routes.
```

Comment:
Can `sources` take ns into consideration? When ns is not provided, the source is supported across all ns's. For example, only routes in ns foo and CRDs in ns bar will be processed:
```
kind: externaldns
metadata:
  name: default
  namespace: my-externaldns-namespace
spec:
...
  sources:
    - foo/route
    - bar/CRD
```

If ns is unspecified, then the source is processed across all ns's.

Reply:
I like this idea. I updated this EP to reflect this improvement. Thanks!